test/natalie/array_test.rb

```diff
@@ -44,6 +44,20 @@
     end
   end
 
+  describe 'hash' do
+    it 'does not collide when array contains duplicate elements' do
+      found_hashes = []
+      (0..100).map do |length| 
+        current_hash = (0..length).to_a.map { |x| 0 }.hash 
+        if found_hashes.find_index current_hash
+          fail
+        else
+          found_hashes << current_hash
+        end 
```

Comment:
`fail` is an alias for `raise`, so I think to fail the spec properly we should use an expectation:

```suggestion
        found_hashes.find_index(current_hash).should be_nil
        found_hashes << current_hash
```